test/unit/org/apache/cassandra/cql3/validation/operations/CreateTest.java

```diff
@@ -365,6 +372,31 @@ public void testKeyspace() throws Throwable
         execute("DROP KEYSPACE testXYZ");
     }
 
+    /**
+     *  Test a warning is thrown on create keyspace with a RF > number of nodes.
+     */
+    @Test
+    public void testCreateKeyspaceRFgtNodesWarns() throws Throwable
```

Comment:
There are other tests checking warnings without it and passing. But I noticed `ant jar test -Dtest.name=CreateTest -Dtest.methods=testCreateKeyspaceRFgtNodesWarns` & variation runs all tests or none, seems broken Did you manage to run it in isolation?

Reply:
I can run that test alone with `ant testsome -Dtest.name=org.apache.cassandra.cql3.validation.operations.CreateTest -Dtest.methods=testCreateKeyspaceRFgtNodesWarns`. That produces a NPE while getting the warnings unless we call `requireNetwork()`. However, as you said there are other tests for warnings that pass without requiring network:
```
ant testsome -Dtest.name=org.apache.cassandra.cql3.ViewTest -Dtest.methods=testClientWarningOnCreate
ant testsome -Dtest.name=org.apache.cassandra.cql3.validation.entities.UFSecurityTest -Dtest.methods=testAmokUDF
ant testsome -Dtest.name=org.apache.cassandra.index.sasi.SASICQLTest -Dtest.methods=testClientWarningOnCreate
```
I'm not sure what is the reason for that difference, maybe it's related to where the call to `ClientWarn.instance.warn` is done?

Reply:
Ok found it. Without the network the test sees 0 nodes and that has been short circuited. With the network we see the node and then the test passes. As the other tests don't rely on the number of nodes to generate a warning they pass without the network. Also I was using `test` instead of `testsome` ops!

Reply:
That makes sense, +1